src/core/packages/saved-objects/migration-server-internal/src/model/model.ts

```diff
@@ -537,27 +537,37 @@ export const model = (currentState: State, resW: ResponseType<AllActionStates>):
   } else if (stateP.controlState === 'CLEANUP_UNKNOWN_AND_EXCLUDED') {
     const res = resW as ExcludeRetryableEsError<ResponseType<typeof stateP.controlState>>;
     if (Either.isRight(res)) {
-      if (res.right.unknownDocs.length) {
+      if (res.right.type === 'cleanup_started') {
+        if (res.right.unknownDocs.length) {
+          logs = [
+            ...stateP.logs,
+            { level: 'warning', message: extractDiscardedUnknownDocs(res.right.unknownDocs) },
+          ];
+        }
+
         logs = [
-          ...stateP.logs,
-          { level: 'warning', message: extractDiscardedUnknownDocs(res.right.unknownDocs) },
+          ...logs,
+          ...Object.entries(res.right.errorsByType).map(([soType, error]) => ({
+            level: 'warning' as const,
+            message: `Ignored excludeOnUpgrade hook on type [${soType}] that failed with error: "${error.toString()}"`,
+          })),
         ];
-      }
 
-      logs = [
-        ...logs,
-        ...Object.entries(res.right.errorsByType).map(([soType, error]) => ({
-          level: 'warning' as const,
-          message: `Ignored excludeOnUpgrade hook on type [${soType}] that failed with error: "${error.toString()}"`,
-        })),
-      ];
-
-      return {
-        ...stateP,
-        logs,
-        controlState: 'CLEANUP_UNKNOWN_AND_EXCLUDED_WAIT_FOR_TASK',
-        deleteByQueryTaskId: res.right.taskId,
-      };
+        return {
+          ...stateP,
+          logs,
+          controlState: 'CLEANUP_UNKNOWN_AND_EXCLUDED_WAIT_FOR_TASK',
+          deleteByQueryTaskId: res.right.taskId,
+        };
+      } else {
```

Comment:
I think it is better to be explicit about the expected state

```suggestion
      } else if (res.right.type === 'cleanup_not_needed') {
```

Perhaps, in an `else` block we can consider throwing using something like `throwBadControlState`

Reply:
Oh that looks better, thanks!